### PR TITLE
fix(levm): op codesize gas

### LINKED ISSUE #1377: fix(levm): opcode CODESIZE
- The gas is consumed after altering the stack. It should be consumed before in case of lack of funds.
- There’s a potential redundant gas consumption check at the beginning of the implementation. Double-check that it is necessary. This check should probably exist in an upper layer.

### PR DESCRIPTION
**Pull request by lima-limon-inc**
**Motivation**

Currently, the CODESIZE operation checks twice to see if there is enough gas. This is redundant. 

There should be only one check BEFORE execution, to avoid executing when there are no more funds left

**Description**

Move check for gas funds before execution.

Closes #1377

**Changes**
- Make the consumption of gas happen before execution for CODESIZE operand
- Add STOP gas cost to the gas_cost file
- Add a small test to check that the VM detects that it ran out of gas

### CODE REVIEW

**Comment by lima-limon-inc** [comment]
Maybe it's worth adding an additional test? Like a `fn test out_of_gas`? Checking for `MaxGasLimitExceeded`? I couldn't find a test for this behavior.

File: crates/vm/levm/src/opcode_handlers/environment.rs [hunk 1]
```diff
@@ -194,22 +194,12 @@ impl VM {
...
     ) -> Result<OpcodeSuccess, VMError> {
-        if self
-            .env
-            .consumed_gas
-            .checked_add(gas_cost::CODESIZE)
-            .ok_or(VMError::OutOfGas(OutOfGasError::ConsumedGasOverflow))?
-            > self.env.gas_limit
-        {
-            return Err(VMError::OutOfGas(OutOfGasError::MaxGasLimitExceeded));
-        }
```
Comment (ilitteri) [inline crates/vm/levm/src/opcode_handlers/environment.rs:205]: @JereSalo is this check made in an upper layer?

Comment (lima-limon-inc) [reply crates/vm/levm/src/opcode_handlers/environment.rs:205]: IINM the `increase_consumed_gas` function  already does that checking. 

Source: 
https://github.com/lambdaclass/ethrex/blob/main/crates/vm/levm/src/vm.rs#L1010-L1023

So, in theory, by calling the `increase_consumed_gas` before anything, that check is done.

Comment (ilitteri) [reply crates/vm/levm/src/opcode_handlers/environment.rs:205]: You're right!

**Review by ilitteri — APPROVED** [review]
LGTM

**Review by maximopalopoli — APPROVED** [review]
LGTM

**Review by JereSalo — APPROVED** [review]
LGTM

**Pull request merged**